File: app/src/main/java/org/digitalcampus/oppia/utils/custom_prefs/AdminMultiSelectListPreference.java
package org.digitalcampus.oppia.utils.custom_prefs;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.ListPreference;
import androidx.preference.MultiSelectListPreference;

public class AdminMultiSelectListPreference extends MultiSelectListPreference implements AdminPreference {

    private OnPreferenceClickListener customOnPreferenceClickListener;

    public AdminMultiSelectListPreference(Context context) {
        super(context);
    }

    public AdminMultiSelectListPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public AdminMultiSelectListPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onClick() {

        if (customOnPreferenceClickListener != null) {
            boolean handled = customOnPreferenceClickListener.onPreferenceClick(this);
            if (handled) {
                return;
            }
        }

        super.onClick();

    }

    @Override
    public void setOnPreferenceClickListener(OnPreferenceClickListener onPreferenceClickListener) {
        customOnPreferenceClickListener = onPreferenceClickListener;
    }

    @Override
    public void onAccessGranted() {
        super.onClick();
    }
}

File: app/src/main/java/org/digitalcampus/oppia/utils/custom_prefs/AdminMultiSelectListPreference.kt
package org.digitalcampus.oppia.utils.custom_prefs

import android.content.Context
import android.util.AttributeSet
import androidx.preference.MultiSelectListPreference

class AdminMultiSelectListPreference : MultiSelectListPreference, AdminPreference {
    private var customOnPreferenceClickListener: OnPreferenceClickListener? = null

    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {}

    override fun onClick() {
        if (customOnPreferenceClickListener != null) {
            val handled = customOnPreferenceClickListener!!.onPreferenceClick(this)
            if (handled) {
                return
            }
        }
        super.onClick()
    }

    override fun setOnPreferenceClickListener(onPreferenceClickListener: OnPreferenceClickListener?) {
        customOnPreferenceClickListener = onPreferenceClickListener
    }

    override fun onAccessGranted() {
        super.onClick()
    }
}